cpp/foxglove-websocket/CMakeLists.txt

```diff
@@ -4,10 +4,17 @@ project(FoxgloveWebSocket CXX)
 find_package(nlohmann_json REQUIRED)
 find_package(websocketpp REQUIRED)
 
+option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
+
 add_library(foxglove_websocket src/base64.cpp src/parameter.cpp src/serialization.cpp src/server_factory.cpp)
+
 target_include_directories(foxglove_websocket PUBLIC include)
+
 target_link_libraries(foxglove_websocket nlohmann_json::nlohmann_json websocketpp::websocketpp)
-set_target_properties(foxglove_websocket PROPERTIES CXX_STANDARD 17 CXX_STANDARD_REQUIRED ON)
+
+set_target_properties(foxglove_websocket PROPERTIES CXX_STANDARD 17
+                                                    CXX_STANDARD_REQUIRED ON
+                                                    POSITION_INDEPENDENT_CODE ${BUILD_SHARED_LIBS})
```

Comment:
Not sure if this is the right way to set export SHARED library; I have no idea how to do it right; if it works it works

Reply:
Addressed via Slack